src/gentropy/dataset/biosample_index.py

```diff
@@ -70,3 +70,25 @@ def merge_indices(
             _df=aggregated_df,
             _schema=BiosampleIndex.get_schema()
             )
+
+    def retain_rows_with_ancestor_id(
+        self: BiosampleIndex,
+        ancestor_ids : list[str]
+        ) -> BiosampleIndex:
+        """Filter the biosample index to retain only rows with the given ancestor IDs.
+
+        Args:
+            ancestor_ids (list[str]): Ancestor IDs to filter on.
+
+        Returns:
+            BiosampleIndex: Filtered biosample index.
+        """
+        # Create a Spark array of ancestor IDs prior to filtering
+        ancestor_ids_array = f.array(*[f.lit(id) for id in ancestor_ids])
```

Comment:
Very good! Given the current situation, when you have a single id to filter on, I would tempted to use `f.array_contains()` method, however that would not scale for a list of identifiers. On the other hand, we hope we'll never have to apply different filters on different datasets... :D